### PR TITLE
feat: error out if more than one package declares the same package name in a workspace

### LINKED ISSUE #3650: `forc build` does not build all workspace members if two members share the same name
When running `forc build` and using a workspace, if members have the same name in their local Forc.toml file, both will not compile. 

In the following example this is the project structure:

```
test_project/
├──  member1/
|     └── src/
|     |     └── main.sw
|     └── Forc.toml
├── member2/
|     └── src/
|     |     └── main.sw
|     └── Forc.toml
└── Forc.toml
```

This is the parent directory Forc.toml file: 

```rust
[workspace]
members = [
  "./member1",
  "./member2",
]
```

Both members of the workspace have the following identical local Forc.toml file:

```rust
[project]
authors = ["Cam Fuel"]
entry = "main.sw"
license = "Apache-2.0"
name = "member"

[dependencies]
```

Despite both being listed as separate members in the workspace, because they have the same name in the local Forc.toml file only one will compile. 

Tested with forc v0.32.2

### PR DESCRIPTION
**Pull request by kayagokalp**
closes #3650

### CODE REVIEW

File: forc-pkg/src/manifest.rs [hunk 1]
```diff
@@ -833,6 +834,34 @@ impl WorkspaceManifest {
                     member_path
                 );
             }
+            let member_manifest_file = PackageManifestFile::from_file(member_path.clone())?;
+            let pkg_name = member_manifest_file.manifest.project.name;
+            if let Some(paths) = pkg_name_to_paths.get_mut(&pkg_name) {
+                paths.push(member_path);
+            } else {
+                pkg_name_to_paths.insert(pkg_name, vec![member_path]);
+            }
```
Comment (mitchmindtree) [inline forc-pkg/src/manifest.rs:843]:
```suggestion
            pkg_name_to_paths.entry(pkg_name).or_default().push(member_path);
```

File: forc-pkg/src/manifest.rs [hunk 2]
```diff
@@ -833,6 +834,34 @@ impl WorkspaceManifest {
...
+        }
+
+        // Check for duplicate pkg name entries in member manifests of this workspace.
+        let duplciate_pkg_lines = pkg_name_to_paths
+            .iter()
+            .filter(|(_, paths)| paths.len() > 1)
+            .map(|(pkg_name, _)| pkg_name)
+            .map(|pkg_name| {
```
Comment (mitchmindtree) [inline forc-pkg/src/manifest.rs:851]:
```suggestion
            .map(|(pkg_name, _)| {
```

File: forc-pkg/src/manifest.rs [hunk 3]
```diff
@@ -833,6 +834,34 @@ impl WorkspaceManifest {
...
+                let duplicate_paths = pkg_name_to_paths
+                    .get(pkg_name)
+                    .expect("missing duplicate paths");
+                format!("{}: {:?}", pkg_name, duplicate_paths)
```
Comment (mitchmindtree) [inline forc-pkg/src/manifest.rs:855]:
```suggestion
                format!("{}: {:#?}", pkg_name, duplicate_paths)
```
^ I _think_ this should print each path on a new line to improve readability a little, but I could be wrong.

**Review by mitchmindtree — COMMENTED** [review]
Looking good! Just some small cleanup then happy to land.

File: forc-pkg/src/manifest.rs (hunk 3 above)
Comment (kayagokalp) [reply forc-pkg/src/manifest.rs:850]: Just tested with and without this change and pretty print is easily 10x better, thanks!

**Review by mitchmindtree — APPROVED** [review]

**Review by JoshuaBatty — APPROVED** [review]

**Pull request merged**